membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -129,10 +127,11 @@ class AccountController(
             subscription.termEndDate,
           )
           result = cancellationEffectiveDate.getOrElse("now").toString
+          catalog <- EitherT.rightT(services.futureCatalog)
```

Comment:
now there's a lot of places we have to flatmap the catalog, I did wonder whether to carry on passing the catalog in to the deserialiser, but instead of using it, just store it in a private field in the Subscription object.  Then when we are reading the old fields (which are now defined on the fly) it could return the values without requiring the catalog as a parameter.